### PR TITLE
pkg/ansible; Fix implementation of watching dependent resources

### PR DESCRIPTION
**Pull request by dymurray**
**Description of the change:**
Prior to this the `watchDependentResources` field in watches.yaml was not being passed along to the runner. We were also setting the wrong values when re-queuing for the owner.

**Motivation for the change:**
Fixes watching of dependent resources for Ansible Operator

In a follow-up PR I will address the corner-case of a namespaced operator trying to watch cluster-scoped resources.

### CODE REVIEW

**Comment by dymurray** [comment]
Travis failure seems to be a transient networking problem. I'm not seeing `make test/markdown` fail on my local machine.

File: pkg/ansible/proxy/proxy.go [hunk 1]
```diff
@@ -189,12 +187,35 @@ func InjectOwnerReferenceHandler(h http.Handler, cMap *ControllerMap) http.Handl
...
+
+			dataClusterScoped := dataMapping.Scope.Name() != meta.RESTScopeNameRoot
+			ownerClusterScoped := ownerMapping.Scope.Name() != meta.RESTScopeNameRoot
+			if !ownerClusterScoped || dataClusterScoped {
+				// add watch for resource
+				err = addWatchToController(owner, cMap, data)
+				if err != nil {
+					m := "could not add watch to controller"
+					log.Error(err, m)
+					http.Error(w, m, http.StatusInternalServerError)
```
Comment (mhrivnak) [inline pkg/ansible/proxy/proxy.go]: I think there needs to be a return statement after this.

Comment (dymurray) [reply pkg/ansible/proxy/proxy.go]: Thanks, fixed.

File: pkg/ansible/proxy/proxy.go [hunk 2]
```diff
@@ -189,12 +188,36 @@ func InjectOwnerReferenceHandler(h http.Handler, cMap *ControllerMap) http.Handl
...
+			ownerMapping, err := restMapper.RESTMapping(schema.GroupKind{Kind: owner.Kind, Group: ownerGV.Group}, ownerGV.Version)
+			if err != nil {
+				m := fmt.Sprintf("could not get rest mapping for: %v", owner)
+				log.Error(err, m)
+				http.Error(w, m, http.StatusInternalServerError)
+				return
+			}
+
+			dataClusterScoped := dataMapping.Scope.Name() != meta.RESTScopeNameRoot
+			ownerClusterScoped := ownerMapping.Scope.Name() != meta.RESTScopeNameRoot
```
Comment (joelanford) [inline pkg/ansible/proxy/proxy.go]: I may be misinterpreting, but `meta.RESTScopeNameRoot` is the name for the cluster scope right (as opposed to `meta.RESTScopeNameNamespace`)? Shouldn't this be `ownerClusterScoped := ownerMapping.Scope.Name() == meta.RESTScopeNameRoot` (i.e. `==`, not `!=`). Same for `dataClusterScoped`.

If so, does that change the logic of the if statement in line 212?
